File: client/src/app/helpers/models/nightChecklist.ts
export interface NightChecklistInterface {
    date: Date;
    glassOfWater: boolean;
    meds: boolean;
    vitamins: boolean;
    washFace: boolean;
    floss: boolean;
    checkEmails: boolean;
    checkTexts: boolean;
    mouthguard: boolean;
    fruits: boolean;
    vegetables: boolean;
    read: boolean;
    wentOutside: boolean;
}

export class NightChecklist {
    date = new Date();
    glassOfWater = false;
    meds = false;
    vitamins = false;
    washFace = false;
    floss = false;
    checkEmails = false;
    checkTexts = false;
    mouthguard = false;
    fruits = false;
    vegetables = false;
    read = false;
    wentOutside = false;

    constructor() {
    }
}

File: API/Models/Entities/NightChecklist.cs
namespace API.Models.Entities {
    public class NightChecklist : Checklist {
        public int NightChecklistID { get; set; }
        public override int UserID { get; set; }
        public override DateOnly Date { get; set; }
        public bool? GlassOfWater { get; set; }
        public bool? Meds { get; set; }
        public bool? Vitamins { get; set; }
        public bool? WashFace { get; set; }
        public bool? Floss { get; set; }
        public bool? Retainer { get; set; }
        public override int GetID() { return NightChecklistID; }
    }
}